<launch>

    <!-- Arguments -->
    <arg name="model" default="$(env TURTLEBOT3_MODEL)" doc="model type [burger, waffle, waffle_pi]"/>
    <arg name="map_file" default="$(find navlab_turtlebot_sim)/maps/flightroom.yaml"/>
    <arg name="open_rviz" default="true"/>
    <arg name="move_forward_only" default="false"/>
    <arg name="remote" default="false"/>
    <arg name="server" default="mocap.local"/>
    <arg name="ns" default="turtlebot3"/>

    <!-- VRPN client -->
    <node pkg="vrpn_client_ros" type="vrpn_client_node" name="vrpn_client_node" output="screen">
        <rosparam subst_value="true">
            server: $(arg server)
            port: 3883
            update_frequency: 100.0
            frame_id: odom
            use_server_time: false
            broadcast_tf: true
            refresh_tracker_frequency: 1.0
        </rosparam>
    </node>

    <!-- Map server -->
    <node pkg="map_server" name="map_server" type="map_server" args="$(arg map_file)"/>

    <!-- static map publisher -->
    <node pkg="tf" type="static_transform_publisher" name="map_odom_publisher" args="0 0 0 0 0 0 1 map odom 100" />

    <!-- rviz -->
    <group if="$(arg open_rviz)">
      <node pkg="rviz" type="rviz" name="rviz" required="true"
            args="-d $(find navlab_turtlebot_sim)/rviz/turtlebot3_navigation.rviz"/>
    </group>


    <!-- TURTLEBOT STUFF -->

    <!-- move_base -->
    <include file="$(find navlab_turtlebot_sim)/launch/move_base.launch">
      <!-- <arg name="ns" value="$(arg ns)" /> -->
      <arg name="model" value="$(arg model)" />
      <arg name="move_forward_only" value="$(arg move_forward_only)"/>
    </include>

    <node pkg="tf" type="static_transform_publisher" name="tf_$(arg ns)" args="0 0 0 0 0 0 1 $(arg ns) base_footprint 100" />

    <!-- Turtlebot3 -->
    <include file="$(find turtlebot3_bringup)/launch/turtlebot3_remote.launch">
      <arg name="model" value="$(arg model)" />
      <!-- <arg name="multi_robot_name" value="$(arg ns)"/> -->
    </include>


</launch>
